control-plane/agents/src/bin/core/controller/scheduling/resources/mod.rs

```diff
@@ -101,6 +102,33 @@ impl PoolItemLister {
             None => vec![],
         }
     }
+    /// Get a list of replicas wrapped as ChildItem, for resize.
+    pub(crate) async fn list_for_resize(registry: &Registry, spec: &VolumeSpec) -> Vec<ChildItem> {
+        let replicas = registry.specs().volume_replicas(&spec.uuid);
+        let state_replicas = registry.replicas().await;
```

Comment:
A volume usually only has a few replicas, probably better to just get their states, ex:

```suggestion
        let mut state_replicas = Vec::with_capacity(replicas.len());
        for replica in &replicas {
            if let Ok(replica) = registry.replica(replica.uuid()).await {
                state_replicas.push(replica);
            }
        }
```